lib/lit/cloud/providers/google.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+require_relative 'base'
+require 'google/cloud/translate'
+
+module Lit::Cloud::Providers
+  # Google Cloud Translation API provider for Lit translation suggestions.
+  #
+  # Configuration:
+  #
+  #   require 'lit/cloud/providers/google'
+  #
+  #   Lit::Cloud.provider = Lit::Cloud::Providers::Google
+  #
+  #   # Service account configuration can be given via a file pointed to by
+  #   # ENV['GOOGLE_TRANSLATE_API_KEYFILE'] (see here:
+  #   # https://cloud.google.com/iam/docs/creating-managing-service-account-keys)
+  #   #
+  #   # Alternatively, the contents of that file can be given as a Ruby hash
+  #   # and passed like the following:
+  #
+  #   Lit::Cloud.configure do |config|
+  #     config.keyfile_hash = {
+  #       'type' => 'service_account',
```

Comment:
I think it should be possible to set all those via ENV variables as well. Consider Heroku or AWS (or anything with more or less ephemeral storage) - setting file without uploading it to repo will be difficult. Setting hash in code will lead to leaking credentials. Maybe ENV variable could store whole json content?

Reply:
I'm going to try coming up with a solution for this. Essentially setting hash in code will not lead to credential leaking if the hash itself uses ENV. But yeah... it would be good to identify the minimal set of google credentials that need to be present for this to work.